bliss/datasets/galsim_galaxies.py

```diff
@@ -106,51 +106,101 @@ def test_dataloader(self):
         self.n_batches = n_batches
         self.batch_size = batch_size
         self.num_workers = num_workers
 
-        self.n_bands = 1
-        self.slen = slen
-        self.background = torch.zeros((self.n_bands, self.slen, self.slen), dtype=torch.float32)
-        self.background[...] = background
-        self.pixel_scale = pixel_scale
+    def __getitem__(self, idx):
+        galaxy_params = self.prior.sample(1, "cpu")
+        return self.decoder.render_galaxy(galaxy_params[0])
+
+    def __len__(self):
+        return self.batch_size * self.n_batches
+
+    def train_dataloader(self):
+        return DataLoader(self, batch_size=self.batch_size, num_workers=self.num_workers)
+
+    def val_dataloader(self):
+        return DataLoader(self, batch_size=self.batch_size, num_workers=self.num_workers)
+
+    def test_dataloader(self):
+        return DataLoader(self, batch_size=self.batch_size, num_workers=self.num_workers)
+
 
+class GalsimGalaxyPrior:
+    def __init__(
+        self,
+        flux_sample: str,
+        min_flux: float,
+        max_flux: float,
+        a_sample: str,
+        alpha: Optional[float] = None,
+        min_a_d: Optional[float] = None,
+        max_a_d: Optional[float] = None,
+        min_a_b: Optional[float] = None,
+        max_a_b: Optional[float] = None,
+        a_concentration: Optional[float] = None,
+        a_loc: Optional[float] = None,
+        a_scale: Optional[float] = None,
+        a_bulge_disk_ratio: Optional[float] = None,
+    ) -> None:
+        self.flux_sample = flux_sample
         self.min_flux = min_flux
         self.max_flux = max_flux
         self.alpha = alpha
+        if self.flux_sample == "pareto":
+            assert self.alpha is not None
 
+        self.a_sample = a_sample
         self.min_a_d = min_a_d
         self.max_a_d = max_a_d
         self.min_a_b = min_a_b
         self.max_a_b = max_a_b
 
-        self.flux_sample = flux_sample
-
-        self.psf = load_psf_from_file(psf_image_file, self.pixel_scale)
+        self.a_concentration = a_concentration
+        self.a_loc = a_loc
+        self.a_scale = a_scale
+        self.a_bulge_disk_ratio = a_bulge_disk_ratio
 
-        if self.flux_sample == "pareto":
-            assert self.alpha is not None
+    def sample(self, total_latent, device):
+        # create galaxy as mixture of Exponential + DeVacauleurs
+        if self.flux_sample == "uniform":
+            total_flux = self._uniform(self.min_flux, self.max_flux, n_samples=total_latent)
+        elif self.flux_sample == "pareto":
+            total_flux = self._draw_pareto_flux(n_samples=total_latent)
+        else:
+            raise NotImplementedError()
+        disk_frac = self._uniform(0, 1, n_samples=total_latent)
+        beta_radians = self._uniform(0, 2 * np.pi, n_samples=total_latent)
+        disk_q = self._uniform(0, 1, n_samples=total_latent)
+        bulge_q = self._uniform(0, 1, n_samples=total_latent)
+        if self.a_sample == "uniform":
```

Comment:
same for the type of size distribution

Reply:
I can add some checks here; better to catch this at init time